3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -12,21 +12,21 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
 // ! Do not change or remove the next two lines
 const rollDie = require('../../helpers/pokerDiceRoller');
 
-function rollDieUntil(wantedValue) {
-  // TODO: rewrite this function using async/await
-  return rollDie().then((value) => {
-    if (value !== wantedValue) {
-      return rollDieUntil(wantedValue);
-    }
-    return value;
-  });
+async function rollDieUntil(wantedValue) {
+  let value = null;
+  while (value !== wantedValue) {
+    value = await rollDie();
+  }
+  return value;
 }
 
-// TODO refactor this function to use try/catch
-function main() {
-  rollDieUntil('ACE')
-    .then((results) => console.log('Resolved!', results))
-    .catch((error) => console.log('Rejected!', error.message));
+async function main() {
+  const ROLL_DIE_UNTIL = 'ACE';
+  try {
+    console.log('Resolved!', await rollDieUntil(ROLL_DIE_UNTIL));
```

Comment:
I guess I'm not getting my point across. I expected something like this:
```suggestion
    const dieValue = await rollDieUntil('ACE');
    console.log('Resolved!', dieValue);
```
The `dieValue` variable explains that what `rollDieUntil('ACE')` return a die value.

The constant `ROLL_DIE_UNTIL` does not add new meaning. It is just a repetition of the function name.